packages/common/src/transports/wiresaw.ts

```diff
@@ -0,0 +1,163 @@
+import { EIP1193RequestFn, Hex, RpcTransactionReceipt, Transport } from "viem";
+import { estimateUserOperationGas } from "./methods/estimateUserOperationGas";
+import { getUserOperationReceipt } from "./methods/getUserOperationReceipt";
+
+type WiresawSendUserOperationResult = {
+  txHash: Hex;
+  userOpHash: Hex;
+};
+
+type WiresawOptions<transport extends Transport> = {
+  wiresaw: transport;
+  fallbackBundler?: Transport;
+  fallbackEth?: Transport;
+};
```

Comment:
just sketching out other name ideas

```ts
type WiresawOptions<transport extends Transport> = {
  /** wiresaw-compatible transport */
  transport: transport;
  /** fallback transport for bundler RPC methods not supported by wiresaw */
  bundlerTransport?: Transport;
  /** fallback transport for all other RPC methods not supported by wiresaw */
  fallbackTransport?: Transport;
};
```

```ts
type WiresawOptions<wiresawTransport extends Transport> = {
  /** wiresaw-compatible transport */
  wiresawTransport: wiresawTransport;
  /** fallback transport for bundler RPC methods not supported by wiresaw */
  bundlerTransport?: Transport;
  /** fallback transport for all other RPC methods not supported by wiresaw */
  defaultTransport?: Transport;
};
```

Reply:
how about:
```ts
type WiresawOptions<transport extends Transport> = {
  /** wiresaw-compatible transport */
  wiresawTransport: transport;
  /** fallback transport for bundler RPC methods */
  fallbackBundlerTransport?: Transport;
  /** fallback transport for all other RPC methods */
  fallbackDefaultTransport?: Transport;
};
```

i'd like to make it clear that the bundler and eth transports are just used for fallback, otherwise it may confuse someone into passing in the wiresaw url into all three since it technically supports all the methods.

Reply:
what does "fallback" mean in this case? maybe that's worth clarifying here in comments 

in viem, fallback means "when transport fails, fall back the request to the next transport" but here we mean a few different things
- for chain ID, block number, block, and fees requests, we prioritize the default transport for request otherwise we use wiresaw
- only for user op receipt, use the bundler transport (again, not really a fallback)
- estimate user op gas _actually_ does a fallback, first wiresaw then bundler
- a catch with actual fallbacks failed requests, falling back to appropriate transport based on req method

I guess "fallback" captures the intent, but the cases where it isn't used as a fallback is a bit weird

Reply:
also internal, so can change later, but still wouldn't mind a rename like you proposed while we're here